apps/papra-server/src/modules/intake-emails/drivers/intake-emails.drivers.models.ts

```diff
@@ -1,12 +1,23 @@
+import type { Organization } from 'better-auth/plugins';
 import type { Config } from '../../config/config.types';
+import type { User } from '../../users/users.types';
 
 export type IntakeEmailsServices = {
   name: string;
-  generateEmailAddress: () => Promise<{ emailAddress: string }>;
-  deleteEmailAddress: ({ emailAddress }: { emailAddress: string }) => Promise<void>;
+  generateEmailAddress: (
+    userHint?: User,
+    organizationHint?: Organization
```

Comment:
The interface for all driver should be as stable a possible, lets mark the user and organization as non-optional to avoid doing per-driver validation, either pass user and org the driver factory (so for each driver) or inject the repositories + userId to avoid unnecessary db fetch for other drivers 

```suggestion
    user: User,
    organization: Organization
```